integrated_channels/sap_success_factors/utils.py

```diff
@@ -93,3 +239,27 @@ def current_time_is_in_interval(start, end):
     interval_start = parse_lms_api_datetime(start or UNIX_MIN_DATE_STRING)
     interval_end = parse_lms_api_datetime(end or UNIX_MAX_DATE_STRING)
     return interval_start <= timezone.now() <= interval_end
+
+
+def parse_language_code(code):
+    """
+    Transform an ISO language code (for example, en-us) into a language name
+    in the format read by SuccessFactors.
+    """
+    if code is None:
+        return 'English'
+    components = code.split('-')
+    language_code = components[0]
+    if 3 > len(components) > 1:
+        country_code = components[1]
+    elif len(components) == 1:
+        country_code = '_'
+    else:
+        raise ValueError('Language codes may only have up to two components. Could not parse: {}'.format(code))
+
+    if language_code not in SUCCESSFACTORS_OCN_LANGUAGE_CODES:
+        LOGGER.warning('Language "%s" is not supported by SuccessFactors. Sending English by default.', language_code)
+        return 'English'
+
+    language_family = SUCCESSFACTORS_OCN_LANGUAGE_CODES[language_code]
+    return language_family.get(country_code, language_family['_'])
```

Comment:
I can definitely make the code take up more lines. 😉

No, we don't need to be concerned. We already handle the case above where the language code isn't in `SUCCESSFACTORS_OCN_LANGUAGE_CODES` by falling back to English. Here, either the country code will be found in the language family (meaning that there's a dialect for that country that has a specific defined name for SAPSF ("Canadian English" for `en-ca`)), or it won't be, and we'll fall through to the base language name (defined by `_`).